src/main/java/com/harena/api/endpoint/rest/mapper/AchatMaterielAuComptantMapper.java

```diff
@@ -0,0 +1,82 @@
+package com.harena.api.endpoint.rest.mapper;
+
+import static java.util.Objects.requireNonNullElse;
+
+import java.util.Objects;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.springframework.stereotype.Component;
+import school.hei.patrimoine.modele.possession.*;
+
+@Component
+@RequiredArgsConstructor
+class AchatMaterielAuComptantMapper
+    implements Mapper<
+        AchatMaterielAuComptant, com.harena.api.endpoint.rest.model.AchatMaterielAuComptant> {
+  private final DeviseMapper deviseMapper;
+  private final PossesionMapper possesionMapper;
+  private final ArgentMapper argentMapper;
+
+  @SneakyThrows
+  private static Object getPrivateFieldValue(String fieldName, Object instance) {
+    var field = Materiel.class.getDeclaredField(fieldName);
+    field.setAccessible(true);
+    var value = field.get(instance);
+    field.setAccessible(false);
+    return value;
+  }
+
+  @Override
+  public com.harena.api.endpoint.rest.model.AchatMaterielAuComptant toRestModel(
+      AchatMaterielAuComptant objectModel) {
+    // The worst things ! cause field: achatCommeGroupe don't have getter
+    GroupePossession groupePossession =
+        (GroupePossession) getPrivateFieldValue("achatCommeGroupe", objectModel);
+    // The worst things ! cause field: possessions don't have getter
```

Comment:
why don't you add a getter to these fields?

Reply:
This class comes from a dependency